### PR TITLE
feat(dedicated.network-security): network security graph

### PR DESCRIPTION
**Pull request by Steffy29**
| Question         | Answer
| ---------------- | ---
| Branch?          | `develop`
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | Fix #UXCT-495
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [x] Only FR translations have been updated
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- [ ] ~Breaking change is mentioned in relevant commits~

## Description

- add traffic chart tab display
- link between row from scrubbing center and traffic chart display
- display graph from API

### CODE REVIEW

File: packages/manager/apps/dedicated/client/app/network-security/network-security.routing.js [hunk 1]
```diff
@@ -10,10 +10,12 @@ export default /* @ngInject */ ($stateProvider) => {
         $translate.instant('network_security_title'),
       goToScrubbingCenter: /* @ngInject */ ($state) => () =>
         $state.go('network-security.scrubbing-center'),
+      goToTraffic: /* @ngInject */ ($state) => () =>
```
Comment (qpavy) [inline packages/manager/apps/dedicated/client/app/network-security/network-security.routing.js:13]: There is not tracking for this action ?

File: packages/manager/apps/dedicated/client/app/network-security/scrubbing-center/scrubbing-center.controller.js [hunk 2]
```diff
@@ -145,4 +121,8 @@ export default class ScrubbingCenterController {
 
     return this.getAllEvents(value.ipBlock);
   }
+
+  showStats(row) {
```
Comment (qpavy) [inline packages/manager/apps/dedicated/client/app/network-security/scrubbing-center/scrubbing-center.controller.js:125]: Use routing resolve to do this action. Doing so, you don't need to make $state accessible to the scope

File: packages/manager/apps/dedicated/client/app/network-security/traffic/traffic.controller.js [hunk 3]
```diff
@@ -0,0 +1,282 @@
...
+import {
+  TRAFFIC_PERIODS,
+  TRAFFIC_PERIOD_LIST,
+  CHART,
+} from './traffic.constant';
+
+export default class TrafficController {
+  /* @ngInject */
+  constructor($http, $state, $translate, Alerter, networkSecurityService) {
+    this.$http = $http;
```
Comment (qpavy) [inline packages/manager/apps/dedicated/client/app/network-security/traffic/traffic.controller.js:13]: You don't need $http as you don't use it in the controller

File: packages/manager/apps/dedicated/client/app/network-security/traffic/traffic.controller.js [hunk 4]
```diff
@@ -0,0 +1,282 @@
...
+    if (!value) {
+      return;
+    }
+    this.subnet = value.ipBlock ? value.ipBlock : value;
+    this.getTraffic();
+  }
+
+  getTraffic() {
+    if (this.$state.params.subnet) {
+      this.subnet = this.$state.params.subnet;
```
Comment (qpavy) [inline packages/manager/apps/dedicated/client/app/network-security/traffic/traffic.controller.js:92]:
```suggestion
      this.subnet = this.getSubnet();
```
retrieve subnet in routing resolve so you don't need to make $state accessible in the scope

File: packages/manager/apps/dedicated/client/app/network-security/traffic/traffic.controller.js [hunk 5]
```diff
@@ -0,0 +1,282 @@
...
+      this.selectedIp = this.subnet;
+    }
+    if (!this.subnet) {
+      return;
+    }
+    this.isLoading = true;
+    this.results = null;
+    let after = '';
+    const before = moment().toISOString();
```
Comment (qpavy) [inline packages/manager/apps/dedicated/client/app/network-security/traffic/traffic.controller.js:101]: If possible, try to not use library like moment
```suggestion
    const currentDate = new Date(); 
    const before = currentDate.toISOString();
```

File: packages/manager/apps/dedicated/client/app/network-security/traffic/traffic.controller.js [hunk 6]
```diff
@@ -0,0 +1,282 @@
...
+    switch (this.period.name) {
+      case this.TRAFFIC_PERIOD_LIST.lastWeek:
+        after = moment()
+          .subtract(7, 'days')
+          .toISOString();
```
Comment (qpavy) [inline packages/manager/apps/dedicated/client/app/network-security/traffic/traffic.controller.js:106]:
```suggestion
        after = new Date();
        after.setDate(after.getDate() - 7);
        after.toISOString();
```

File: packages/manager/apps/dedicated/client/app/network-security/traffic/traffic.controller.js [hunk 7]
```diff
@@ -0,0 +1,282 @@
...
+        break;
+      case this.TRAFFIC_PERIOD_LIST.last2weeks:
+        after = moment()
+          .subtract(14, 'days')
+          .toISOString();
```
Comment (qpavy) [inline packages/manager/apps/dedicated/client/app/network-security/traffic/traffic.controller.js:111]:
```suggestion
        after = new Date();
        after.setDate(after.getDate() - 14);
        after.toISOString();
```

File: packages/manager/apps/dedicated/client/app/network-security/traffic/traffic.controller.js [hunk 8]
```diff
@@ -0,0 +1,282 @@
...
+        break;
+      default:
+        after = moment()
+          .subtract(1, 'days')
+          .toISOString();
```
Comment (qpavy) [inline packages/manager/apps/dedicated/client/app/network-security/traffic/traffic.controller.js:116]:
```suggestion
        after = new Date();
        after.setDate(after.getDate() - 1);
        after.toISOString();
```

File: packages/manager/apps/dedicated/client/app/network-security/traffic/traffic.controller.js [hunk 9]
```diff
@@ -0,0 +1,282 @@
...
+    this.series = [];
+    this.data = [];
+
+    if (this.isStackable) {
+      this.options.scales.yAxes[0].stacked = true;
+    } else {
+      delete this.options.scales.yAxes[0].stacked;
+    }
+
+    this.labels = map(get(this.results, 'timestamps'), (value) => value);
```
Comment (qpavy) [inline packages/manager/apps/dedicated/client/app/network-security/traffic/traffic.controller.js:214]: Try to use native es6
```suggestion
    this.labels = this.results?.timestamps?.map((value) => value)
```

**Review by qpavy — CHANGES_REQUESTED** [review]

File: packages/manager/apps/dedicated/client/app/network-security/scrubbing-center/scrubbing-center.controller.js [hunk 10]
```diff
@@ -2,41 +2,27 @@ import { PERIODS, PERIOD_LIST } from './scrubbing-center.constant';
 
 export default class ScrubbingCenterController {
   /* @ngInject */
-  constructor($http, $translate, Alerter, scrubbingCenterService) {
+  constructor($http, $state, $translate, Alerter, networkSecurityService) {
     this.$http = $http;
+    this.$state = $state;
```
Comment (qpavy) [inline packages/manager/apps/dedicated/client/app/network-security/scrubbing-center/scrubbing-center.controller.js:7]: see https://github.com/ovh/manager/pull/10160/files#r1369865288

File: packages/manager/apps/dedicated/client/app/network-security/traffic/traffic.controller.js [hunk 11]
```diff
@@ -0,0 +1,282 @@
...
+  TRAFFIC_PERIODS,
+  TRAFFIC_PERIOD_LIST,
+  CHART,
+} from './traffic.constant';
+
+export default class TrafficController {
+  /* @ngInject */
+  constructor($http, $state, $translate, Alerter, networkSecurityService) {
+    this.$http = $http;
+    this.$state = $state;
```
Comment (qpavy) [inline packages/manager/apps/dedicated/client/app/network-security/traffic/traffic.controller.js:14]: see https://github.com/ovh/manager/pull/10160/files#r1369871781

File: packages/manager/apps/dedicated/client/app/network-security/traffic/traffic.controller.js [hunk 12]
```diff
@@ -0,0 +1,282 @@
...
+    );
+    this.series.push(
+      this.$translate.instant('network_security_dashboard_legend_red'),
+    );
+    if (!this.isPPs) {
+      // Get unit from max value from table
+      let maxPassedValue = 0;
+      let maxDroppedValue = 0;
+      let unitIndex = 0;
+      get(this.results, 'bps.passed').forEach((val) => {
```
Comment (qpavy) [inline packages/manager/apps/dedicated/client/app/network-security/traffic/traffic.controller.js:227]: see https://github.com/ovh/manager/pull/10160/files#r1369888340

File: packages/manager/apps/dedicated/client/app/network-security/traffic/traffic.controller.js [hunk 13]
```diff
@@ -0,0 +1,282 @@
...
+        map(get(this.results, 'bps.dropped'), (value) =>
+          TrafficController.formatBits(parseInt(value, 10), unitIndex),
+        ),
+      );
+    } else {
+      this.options.scales.yAxes[0].scaleLabel.labelString = this.$translate.instant(
+        'network_security_dashboard_pps',
+      );
+
+      this.data.push(map(get(this.results, 'pps.passed'), (value) => value));
```
Comment (qpavy) [inline packages/manager/apps/dedicated/client/app/network-security/traffic/traffic.controller.js:263]: see https://github.com/ovh/manager/pull/10160/files#r1369888340

File: packages/manager/apps/dedicated/client/app/network-security/traffic/traffic.controller.js [hunk 14]
```diff
@@ -0,0 +1,282 @@
...
+      this.data.push(map(get(this.results, 'pps.dropped'), (value) => value));
```
Comment (qpavy) [inline packages/manager/apps/dedicated/client/app/network-security/traffic/traffic.controller.js:264]: see https://github.com/ovh/manager/pull/10160/files#r1369888340

File: packages/manager/apps/dedicated/client/app/network-security/traffic/traffic.controller.js (hunk 9 above)
Comment (ghyenne) [inline packages/manager/apps/dedicated/client/app/network-security/traffic/traffic.controller.js:214]:
```suggestion
    this.labels = this.results?.timestamps.map((value) => value);
```

File: packages/manager/apps/dedicated/client/app/network-security/traffic/traffic.controller.js [hunk 15]
```diff
@@ -0,0 +1,282 @@
...
+      get(this.results, 'bps.passed').forEach((val) => {
+        if (parseInt(val, 10) > maxPassedValue) {
+          maxPassedValue = parseInt(val, 10);
+        }
+      });
+      get(this.results, 'bps.dropped').forEach((val) => {
+        if (parseInt(val, 10) > maxDroppedValue) {
+          maxDroppedValue = parseInt(val, 10);
+        }
+      });
```
Comment (ghyenne) [inline packages/manager/apps/dedicated/client/app/network-security/traffic/traffic.controller.js:236]:
```suggestion
      this.results?.bps.passed.forEach((val) => {
        if (parseInt(val, 10) > maxPassedValue) {
          maxPassedValue = parseInt(val, 10);
        }
      });
      this.results?.bps.dropped.forEach((val) => {
        if (parseInt(val, 10) > maxDroppedValue) {
          maxDroppedValue = parseInt(val, 10);
        }
      });
```

File: packages/manager/apps/dedicated/client/app/network-security/traffic/traffic.html [hunk 16]
```diff
@@ -0,0 +1,136 @@
...
+    <div data-ng-if="$ctrl.results">
+        <div class="d-flex justify-content-start">
+            <div class="oui-box oui-box_light mr-4">
+                <h4
+                    class="oui-box__heading"
+                    data-translate="network_security_dashboard_number_of_events"
+                ></h4>
+                <p class="oui-heading_5 text-center oui-color-p-500">
+                    {{ $ctrl.numberOfEvents }}
+                </p>
```
Comment (ghyenne) [inline packages/manager/apps/dedicated/client/app/network-security/traffic/traffic.html:65]:
```suggestion
                <p class="oui-heading_5 text-center oui-color-p-500" data-ng-bind="$ctrl.numberOfEvents">
                </p>
```

File: packages/manager/apps/dedicated/client/app/network-security/network-security.routing.js (hunk 1 above)
Comment (Steffy29) [reply packages/manager/apps/dedicated/client/app/network-security/network-security.routing.js:13]: Tracking is not set yet.

File: packages/manager/apps/dedicated/client/app/network-security/traffic/traffic.controller.js (hunk 4 above)
Comment (Steffy29) [reply packages/manager/apps/dedicated/client/app/network-security/traffic/traffic.controller.js:92]: I would like but how I can get this value into the controller from the routing?

**Review by qpavy — APPROVED** [review]

**Review by ghyenne — APPROVED** [review]

**Pull request merged**